drivers/wifi/eswifi/eswifi_core.c

```diff
@@ -447,13 +441,163 @@ static int eswifi_mgmt_connect(struct device *dev,
 		eswifi->sta.security = ESWIFI_SEC_WPA2_MIXED;
 		break;
 	default:
-		LOG_ERR("Unsupported security type %d", params->security);
-		eswifi_unlock(eswifi);
 		return -EINVAL;
 	}
 
-	eswifi->req = ESWIFI_REQ_CONNECT;
-	k_work_submit_to_queue(&eswifi->work_q, &eswifi->request_work);
+	if (params->channel == WIFI_CHANNEL_ANY) {
+		eswifi->sta.channel = 0;
+	} else {
+		eswifi->sta.channel = params->channel;
+	}
+
+	return 0;
+}
+
+static int eswifi_mgmt_connect(struct device *dev,
+			       struct wifi_connect_req_params *params)
+{
+	struct eswifi_dev *eswifi = dev->driver_data;
+	int err;
+
+	LOG_DBG("");
+
+	eswifi_lock(eswifi);
+
+	err = __eswifi_sta_config(eswifi, params);
+	if (!err) {
+		eswifi->req = ESWIFI_REQ_CONNECT;
+		k_work_submit_to_queue(&eswifi->work_q,
+				       &eswifi->request_work);
+	}
+
+	eswifi_unlock(eswifi);
+
+	return err;
+}
+
+#if defined(CONFIG_NET_IPV4)
+static int eswifi_mgmt_ap_enable(struct device *dev,
+				 struct wifi_connect_req_params *params)
+{
+	struct eswifi_dev *eswifi = dev->driver_data;
+	struct net_if_ipv4 *ipv4 = eswifi->iface->config.ip.ipv4;
+	struct net_if_addr *unicast = NULL;
+	int err = -EIO, i;
+
+	LOG_DBG("");
+
+	eswifi_lock(eswifi);
+
+	err = __eswifi_sta_config(eswifi, params);
+	if (err) {
+		eswifi_unlock(eswifi);
+		return err;
+	}
+
```

Comment:
Shouldn't you have a way to verify that ap mode is already enabled? (So -EALREADY would be a possible error. user would need to disable before re-enabling it for a different config)

Reply:
Yes, I'm going to send a new version.